Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -309,4 +312,244 @@ theorem psum_isSymmetric (n : ℕ) : IsSymmetric (psum σ R n) := rename_psum _
+    by_cases hji : j = i
+    · rw [hji, Exponent.one]
+      simp only [ite_true, hi]
+    · have : exp s j = 0 := by
+        obtain (h0 | h1) := (this j)
+        · assumption
+        · exfalso
+          have : 2 <= ∑ i, s.exp i := by
+            let f : Finset σ := {i, j}
+            have : f ⊆ univ := subset_univ f
+            have that : ∑ k in f, s.exp k = 2 := by
+              simp only [Finset.sum_insert]
+              have : ∀ k ∈ f, s.exp k = 1 := by
+                intro k hk
+                have : k = i ∨ k = j := by
+                  simp only [Finset.mem_insert] at hk
+                  obtain (ki | kj) := hk
+                  · left
+                    exact ki
+                  · right
+                    exact mem_singleton.mp kj
+                obtain (ki | kj) := this
+                · rw [ki]
+                  exact hi
+                · rw [kj]
+                  exact h1
+              have f2 : f.card = 2 := by
+                simp only [Finset.card_insert_of_not_mem, Finset.card_singleton, hji]
+                rw [Finset.card_eq_two]
+                use i, j
+                constructor
+                symm
+                simp only [ne_eq, hji]
+                trivial
+                rfl
+              have : ∑ k in f, s.exp k = ∑ k in f, 1 := by
+                apply Finset.sum_congr
+                rfl
+                exact this
+              rw [this]
+              simp only [sum_const, f2, smul_eq_mul, mul_one]
+            rw [←that]
+            exact Finset.sum_le_sum_of_subset this
+          linarith
+      rw [this]
+      simp only [Exponent.one]
+      split
+      tauto
+      rfl
+    intro j
+    by_cases hji : j = i
+    rw [hji]
+    simp only [implies_true]
+    intro h0
+    exfalso
+    rw [←h0] at hi
+    have : exp (one σ j) i = 0 := by simp only [Exponent.one]; split; tauto; rfl
+    linarith
+  · exfalso
+    have : ∀ i : σ, s.exp i = 0 := by
+      intro i
+      obtain (h0 | h1) := (this i)
+      · assumption
+      · exfalso
+        exact exists_i ⟨i, h1⟩
+    have : ∑ i, s.exp i = 0 := by
+      simp only [this, Finset.sum_const_zero]
+    linarith
+  done
+
+def Exponent.fintype_toExponent : (σ → Exponent σ 1) := (fun i => ⟨fun j => if i = j then 1 else 0, by simp only [sum_ite_eq, mem_univ, ite_true]⟩)
+
+lemma Exponent.fintype_toExponent_bijective : Function.Bijective (Exponent.fintype_toExponent σ) := by
+  rw [Function.bijective_iff_existsUnique]
+  exact Exponent.one_sum_singleton σ
+
+lemma Exponent.fintype_toExponent_equiv : σ ≃ Exponent σ 1 := by
+  exact Equiv.ofBijective _ (Exponent.fintype_toExponent_bijective σ)
+  done
+
+lemma Exponent.exponentCard_one : Fintype.card (Exponent σ 1) = Fintype.card σ := by
+  apply Fintype.card_congr
+  symm
+  exact Exponent.fintype_toExponent_equiv _
+  done
+
+end Exponent
+
+section CompleteHomogeneousSymmetric
+
+open Classical
+
+variable (σ R) [CommSemiring R] [CommSemiring S] [Fintype σ] [Fintype τ]
```

Comment:
You already have these lines. You may need to move them before some section, but I think you need them only once.